### PR TITLE
precompile some headers

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

**Review by martun — APPROVED** [review]

**Review by martun — APPROVED** [review]

**Review by martun — APPROVED** [review]

File: flake.nix [hunk 1]
```diff
@@ -47,7 +47,7 @@
           });
           crypto3-clang-debug = (pkgs.callPackage ./crypto3.nix {
             stdenv = pkgs.llvmPackages_19.stdenv;
-            runTests = false;
+            runTests = true;
```
Comment (akokoshn) [inline flake.nix:50]: why don't use `crypto3-clang-debug-tests`

File: parallel-crypto3/libs/parallel-math/test/CMakeLists.txt [hunk 2]
```diff
@@ -35,7 +35,7 @@ macro(define_math_test name)
                                ${Boost_INCLUDE_DIRS})
 
     set_target_properties(${test_name} PROPERTIES CXX_STANDARD 17)
-
+#    target_precompile_headers(${full_test_name} REUSE_FROM _cm_internal_tests-actor-math-test)
```
Comment (akokoshn) [inline parallel-crypto3/libs/parallel-math/test/CMakeLists.txt:38]: remove comment

Comment (AndreyMlashkin) [reply parallel-crypto3/libs/parallel-math/test/CMakeLists.txt]: done

File: flake.nix (hunk 1 above)
Comment (AndreyMlashkin) [reply flake.nix:50]: yes, indeed, reverted

File: parallel-crypto3/benchmarks/polynomial_dfs_benchmark.cpp [hunk 3]
```diff
@@ -0,0 +1,215 @@
+//---------------------------------------------------------------------------//
```
Comment (martun) [inline parallel-crypto3/benchmarks/polynomial_dfs_benchmark.cpp:1]: Was this change intentional?

Comment (AndreyMlashkin) [reply parallel-crypto3/benchmarks/polynomial_dfs_benchmark.cpp]: Yes, I moved the tests to match the crypto3 layout. It also has benchmarks on the top level

**Review by martun — APPROVED** [review]
